packages/fx-core/src/plugins/resource/bot/v2/azureHosting.ts

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Inputs, ResourceTemplate, Void } from "@microsoft/teamsfx-api";
+import { BicepTemplate, Context } from "@microsoft/teamsfx-api/build/v2";
+import * as fs from "fs-extra";
+import path from "path";
+import { BotHostTypes, generateBicepFromFile } from "../../../../common";
+import { ArmTemplateResult } from "../../../../common/armInterface";
+import { Bicep } from "../../../../common/constants";
+import { getTemplatesFolder } from "../../../../folder";
+import { getActivatedV2ResourcePlugins } from "../../../solution/fx-solution/ResourcePluginContainer";
+import { NamedArmResourcePluginAdaptor } from "../../../solution/fx-solution/v2/adaptor";
+import { HostTypes, PluginBot } from "../resources/strings";
+import { BicepConfigs } from "./botSolution";
+import * as utils from "../utils/common";
+
+export interface BicepFiles {
+  provisionOrchestrationFile: string;
+  provisionModuleFile: string;
+  configOrchestrationFile?: string;
+  configModuleFile?: string;
+}
+
+export interface BicepModules {
+  provisionOrchestration: string;
+  provisionModule: string;
+  configOrchestration?: string;
+  configModule?: string;
+}
+
+async function generateBicep(
+  bicepFiles: BicepFiles,
+  generate: (filename: string) => Promise<string>
+): Promise<BicepModules> {
+  let filenames = [bicepFiles.provisionOrchestrationFile, bicepFiles.provisionModuleFile];
+  if (bicepFiles.configModuleFile && bicepFiles.configOrchestrationFile) {
+    filenames = filenames.concat([bicepFiles.configOrchestrationFile, bicepFiles.configModuleFile]);
+  }
+
+  const modules = await Promise.all(filenames.map((filename) => generate(filename)));
+
+  return {
```

Comment:
If we still need to use index here, maybe BicepFiles interface is not well used. Let's offline to think about better solution for it.